src/python/grpcio_tests/tests/interop/server_ssl_cert_reload_test/_test.py

```diff
@@ -0,0 +1,363 @@
+    open(_get_abs_path('cert_hier_1/intermediate/certs/intermediate.cert.pem')).read(),
+    ])
+
+client_key_2_pem = \
+    open(_get_abs_path('cert_hier_2/intermediate/private/client.key.pem')).read()
+client_chain_2_pem = '\n'.join([
+    open(_get_abs_path('cert_hier_2/intermediate/certs/client.cert.pem')).read(),
+    open(_get_abs_path('cert_hier_2/intermediate/certs/intermediate.cert.pem')).read(),
+    ])
+
+server_key_1_pem = \
+    open(_get_abs_path('cert_hier_1/intermediate/private/localhost-1.key.pem')).read()
+server_cert_1_pem = '\n'.join([
+    open(_get_abs_path('cert_hier_1/intermediate/certs/localhost-1.cert.pem')).read(),
+    open(_get_abs_path('cert_hier_1/intermediate/certs/intermediate.cert.pem')).read(),
+])
+
+server_key_2_pem = \
+    open(_get_abs_path('cert_hier_2/intermediate/private/localhost-1.key.pem')).read()
+server_cert_2_pem = '\n'.join([
+    open(_get_abs_path('cert_hier_2/intermediate/certs/localhost-1.cert.pem')).read(),
+    open(_get_abs_path('cert_hier_2/intermediate/certs/intermediate.cert.pem')).read(),
+])
+
+
+class CertConfigFetcher(object):
+  def __init__(self, ca_1_pem, my_key_1_pem, my_cert_1_pem,
+               ca_2_pem, my_key_2_pem, my_cert_2_pem,
+               switch_cert_on_client_num):
+    self.client_num = -1
+    self.ca_2_pem = ca_2_pem
+    self.my_key_2_pem = my_key_2_pem
+    self.my_cert_2_pem = my_cert_2_pem
+    self.switch_cert_on_client_num = switch_cert_on_client_num
+    self.cert_config = grpc.ssl_server_certificate_config(
+        [(my_key_1_pem, my_cert_1_pem)],
+        root_certificates=ca_1_pem,
+      )
+    # we simulate cb failures at 2 and 3
+    assert self.switch_cert_on_client_num > 3
+
+  def __call__(self):
+    self.client_num += 1
+    if self.client_num == 0:
+      # this is actually during server initialzation
+      return self.cert_config
+    else:
+      if self.client_num in (2, 3):
+        raise ValueError('just an error for fun, should not affect the test')
+
+      if self.client_num != self.switch_cert_on_client_num:
+        return None
+      else:
+        # do the switch
+        return grpc.ssl_server_certificate_config(
+          [(self.my_key_2_pem, self.my_cert_2_pem)],
+          root_certificates=self.ca_2_pem,
+        )
+
+
+class _ServerSSLCertReloadTestBase(unittest.TestCase):
+    def setUp(self):
+        raise NotImplementedError()
+
+    def tearDown(self):
+        raise NotImplementedError()
+
+    def _perform_rpc(self, client_stub, expect_success):
+        if expect_success:
+            methods.TestCase.EMPTY_UNARY.test_interoperability(client_stub, None)
+        else:
+            with self.assertRaises(grpc.RpcError) as exc:
+                methods.TestCase.EMPTY_UNARY.test_interoperability(client_stub, None)
+            assert exc.exception.code() == grpc.StatusCode.UNAVAILABLE
+            assert exc.exception.details() == 'Connect Failed', exc.exception.details()
+        return
```

Comment:
What does this return statement accomplish?

Reply:
will remove

Reply:
done